src/test/java/teammates/sqllogic/core/AccountsLogicTest.java

```diff
@@ -34,11 +41,52 @@ public class AccountsLogicTest extends BaseTestCase {
 
     private NotificationsLogic notificationsLogic;
 
+    private UsersLogic usersLogic;
+
+    private Course course;
+
     @BeforeMethod
     public void setUpMethod() {
         accountsDb = mock(AccountsDb.class);
         notificationsLogic = mock(NotificationsLogic.class);
-        accountsLogic.initLogicDependencies(accountsDb, notificationsLogic);
+        usersLogic = mock(UsersLogic.class);
+        accountsLogic.initLogicDependencies(accountsDb, notificationsLogic, usersLogic);
+
+        course = new Course("course-id", "course-name", Const.DEFAULT_TIME_ZONE, "institute");
+    }
+
+    @Test
+    public void testDeleteAccount_accountExists_success() {
+        Account account = generateTypicalAccount();
+        String googleId = account.getGoogleId();
+
+        when(accountsLogic.getAccountForGoogleId(googleId)).thenReturn(account);
+
+        accountsLogic.deleteAccount(googleId);
+
+        verify(accountsDb, times(1)).deleteAccount(account);
+    }
+
+    @Test
+    public void testDeleteAccountCascade_googleIdExists_success() {
+        Account account = generateTypicalAccount();
+        String googleId = account.getGoogleId();
+        List<User> users = new ArrayList<>();
+
+        for (int i = 0; i < 2; ++i) {
+            users.add(getTypicalInstructor());
+            users.add(getTypicalStudent());
+        }
+
+        when(usersLogic.getAllUsersByGoogleId(googleId)).thenReturn(users);
+        when(accountsLogic.getAccountForGoogleId(googleId)).thenReturn(account);
```

Comment:
In future, can apply the following heuristic. Try not to mock other methods in same class unless there's a reason for it (i.e. other method is slow to run, method has significant complexity etc.). Ok to treat the class as a unit most of the time. Adding too many mocks runs the risk of testing Mockito rather than the behaviour of your function if you are not careful and missing some behaviour that can only be exercised by testing methods together.

Reply:
Alright got it! Thank you for sharing this! I thought we should try not to bring in other actual classes when we're testing the current unit itself. Will take note of this. In this case, we could just create all the Users and run `getAllUsersByGoogleId`? I'll try that out. 👍🏻 Thanks samuel!

Reply:
referring to you mocking get account in accounts logic itself.

Reply:
Ah okay I understand! I could change that in the next PR.